src/tests/functional/plugin/gpu/single_layer_tests/dynamic/unique.cpp

```diff
@@ -0,0 +1,170 @@
+            result << "}_";
+        }
+
+        if (!std::get<0>(flatOrAxis)) {
+            result << "axis=" << std::get<1>(flatOrAxis) << "_";
+        } else {
+            result << "flattened"
+                   << "_";
+        }
+        result << "sorted=" << (sorted ? "True" : "False") << "_";
+        result << "dataPrc=" << dataPrecision;
+
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        std::vector<InputShape> inputShapes;
+        std::tuple<bool, int> flatOrAxis;
+        bool sorted, flattened;
+        int axis;
+        ElementType dataPrecision;
+
+        std::tie(inputShapes, flatOrAxis, sorted, dataPrecision) = this->GetParam();
+        targetDevice = CommonTestUtils::DEVICE_GPU;
+        init_input_shapes(inputShapes);
+        flattened = std::get<0>(flatOrAxis);
+
+        auto params = ngraph::builder::makeDynamicParams(dataPrecision, inputDynamicShapes);
+        params[0]->set_friendly_name("data");
+        auto paramOuts =
+            ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ov::op::v0::Parameter>(params));
+        std::shared_ptr<ov::Node> uniqueNode;
+        if (flattened) {
+            uniqueNode = std::make_shared<ov::op::v10::Unique>(paramOuts[0], sorted);
+        } else {
+            axis = std::get<1>(flatOrAxis);
+            uniqueNode = std::make_shared<ov::op::v10::Unique>(
+                paramOuts[0],
+                ov::op::v0::Constant::create(ov::element::i64, ov::Shape({1}), {axis}),
+                sorted);
+        }
+
+        // Need to create results for all outputs
+        ngraph::ResultVector results;
+        for (auto i = 0U; i < uniqueNode->get_output_size(); ++i) {
+            results.push_back(std::make_shared<ngraph::opset1::Result>(uniqueNode->output(i)));
+        }
+
+        function = std::make_shared<ngraph::Function>(results, params, "Unique");
+    }
+
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        for (size_t i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::Tensor tensor;
+
+            if (funcInput.get_node()->get_friendly_name() == "data") {
+                int32_t range = std::accumulate(targetInputStaticShapes[0].begin(),
+                                                targetInputStaticShapes[0].end(),
+                                                1,
+                                                std::multiplies<size_t>());
+                tensor = utils::create_and_fill_tensor(funcInput.get_element_type(),
+                                                       targetInputStaticShapes[0],
+                                                       range,
+                                                       -range / 2,
+                                                       1);
+            }
```

Comment:
I think using only random data for this kind of operation is not sufficient. Could you please add a gpu unittest with some duplicated data? Or could you add such data with dupilication here?

Reply:
I have copied the same generation function from CPU tests, but I agree that it is better to have a deterministic test for such an operation. I will add not random unit tests.

Reply:
Added unit tests